include/swift/AST/DiagnosticsSema.def

```diff
@@ -7977,7 +7977,13 @@ ERROR(lifetime_dependence_cannot_infer_ambiguous_candidate, none,
        ())
 ERROR(lifetime_dependence_immortal_conflict_name, none,
       "conflict between the parameter name and immortal keyword", ())
-
+ERROR(lifetime_dependence_function_type, none,
+      "Lifetime dependencies on function types are not yet supported",
+      ())
+ERROR(lifetime_dependence_function_type_conversion, none,
+      "Conversion between function types with different lifetime dependencies "
+      "are not allowed",
```

Comment:
```suggestion
      "Conversions between function types with different lifetime dependencies "
      "are not allowed",
```
?

Reply:
Could this message follow a “cannot pass” template and instead of “different lifetime dependencies” actually name them? We could add a more general explanation as a note.

Reply:
Done. This is now following "cannot convert value..." template